perf(ast)!: revert: reduce byte size of `TaggedTemplateExpression::quasi` by `Boxing` it (#5679)

overlookmotel opened the pull request:
As discussed in #5601, there is little benefit to reducing type sizes for the sake of it. In this case, the `quasi` field in not in an `Option`, so putting it in a `Box` does not save data, only moves where that data is stored, and introduces pointer-indirection.

overlookmotel commented:
* **#5715** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5715?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
I believe the benchmark changes to be noise. In particular the `transformer[cal.com.tsx]` benchmark change is clearly spurious - the negative all appears to be due to a hashmap reallocating at the start of transform.

Boshen commented:
> As discussed in #5601, there is little benefit to reducing type sizes for the sake of it. In this case, the `quasi` field in not in an `Option`, so putting it in a `Box` does not save data, only moves where that data is stored, and introduces pointer-indirection.

We need a "specification" of what to box and what not.

overlookmotel committed:
perf(ast)!: revert: reduce byte size of `TaggedTemplateExpression::quasi` by `Boxing` it (#5679) (#5715)

As discussed in #5601, there is little benefit to reducing type sizes for the sake of it. In this case, the `quasi` field in not in an `Option`, so putting it in a `Box` does not save data, only moves where that data is stored, and introduces pointer-indirection.
